packages/react/src/components/checkbox/checkbox.tsx

```diff
@@ -17,11 +19,14 @@ export const Checkbox = withRef(
     ref: CheckboxProps['ref']
   ): JSX.Element => {
     const { disabled, touched } = useField();
+    const autoId = useAutoId('castor_checkbox');
+    const id = externalId || autoId;
```

Comment:
We should probably use a Nullish Coalescing operator instead so that `0` is a valid `id`.

Also, if they do `id=""` I suppose we let them override the `autoId` too?

```suggestion
    const id = externalId ?? autoId;
```

Reply:
Actually even `id={null}` should be able to override the `autoId`, I think.

Maybe we just default the property value?

```ts
let count = 0;

const Checkbox = ({ id = `castor_checkbox_${++count}` }) => // ...
```

Reply:
Originally it was [implemented like such](https://github.com/onfido/castor/pull/267/commits/65d002f248780398fad931277d2860c3847f6b76), however was changed to [internal state management](https://github.com/onfido/castor/pull/267/commits/bde7d3361d3306f6ca8dac591bb2abf03a0d2270).

TBH I can't remember why, and this seems to be fine... Unless I'm missing something.

Reply:
The count is going to increment on every render rather than on every component instance.

There shouldn't be any practical difference, other than the attributes changing a lot more.